packages/dds/tree/src/shared-tree/treeApi.ts

```diff
@@ -40,22 +54,124 @@ export interface TreeApi extends TreeNodeApi {
+	 *
+	 * Local change events will be emitted for each change as the transaction is being applied.
+	 * If the transaction function throws an error then the transaction will be automatically rolled back (discarding any changes made to the tree so far) before the error is propagated up from this function.
+	 * If the transaction is rolled back, a corresponding change event will also be emitted for the rollback.
+	 */
+	<TNode extends TreeNode>(node: TNode, transaction: (node: TNode) => void): void;
+	/**
+	 * Apply one or more edits to the tree as a single atomic unit.
+	 * @param tree - The tree which will be edited by the transaction
+	 * @param transaction - The function to run as the body of the transaction.
+	 * This function is passed the root of the tree.
+	 * @remarks
+	 * All of the changes in the transaction are applied synchronously and therefore no other changes (either from this client or from a remote client) can be interleaved with those changes.
+	 * Note that this is guaranteed by Fluid for any sequence of changes that are submitted synchronously, whether in a transaction or not.
+	 * However, using a transaction has the following additional consequences:
+	 * - If reverted (e.g. via an "undo" operation), all the changes in the transaction are reverted together.
+	 * - The internal data representation of a transaction with many changes is generally smaller and more efficient than that of the changes when separate.
+	 *
+	 * Local change events will be emitted for each change as the transaction is being applied.
+	 * If the transaction function throws an error then the transaction will be automatically rolled back (discarding any changes made to the tree so far) before the error is propagated up from this function.
+	 * If the transaction is rolled back, a corresponding change event will also be emitted for the rollback.
+	 */
+	<TView extends TreeView<ImplicitFieldSchema>>(
+		tree: TView,
+		transaction: (root: TView["root"]) => void,
 	): void;
 	/**
 	 * Apply one or more edits to the tree as a single atomic unit.
 	 * @param node - The node that will be passed to `transaction`.
 	 * This is typically the root node of the subtree that will be modified by the transaction.
 	 * @param transaction - The function to run as the body of the transaction.
 	 * This function is passed the provided `node`.
-	 * At any point during the transaction, the function may return the value `"rollback"` to abort the transaction and discard any changes it made so far.
 	 * @param preconditions - An optional list of {@link TransactionConstraint | constraints} that are checked just before the transaction begins.
 	 * If any of the constraints are not met when `runTransaction` is called, it will throw an error.
 	 * If any of the constraints are not met when this transaction is sequenced (on this client or other clients), the transaction will not be run.
```

Comment:
edits are not "sequenced" on a client, and I think this is confusing. we also might not want to introduce sequencing to the customer. instead, maybe say "are not met when the changes in the transaction are applied (on any client) after being ordered by the service..."

Reply:
Github messed up the line number but I'm assuming you are talking about the line three lines below. I changed it to

"If any of the constraints are not met after the transaction has been ordered by the service, it will be rolled back on this client and ignored by all other clients."

Let me know what you think.